datafusion/physical-expr/src/aggregate/count_distinct.rs

```diff
@@ -30,30 +30,31 @@ use datafusion_common::ScalarValue;
 use datafusion_common::{DataFusionError, Result};
 use datafusion_expr::Accumulator;
 
-type DistinctScalarValues = ScalarValue;
+#[derive(Debug, PartialEq, Eq, Hash, Clone)]
+struct DistinctScalarValues(Vec<ScalarValue>);
```

Comment:
The downside is some performance loss / increased memory usage for single expressions.